core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -190,18 +176,16 @@ final ClientTransport obtainActiveTransport() {
       }
       if (state.getState() == IDLE) {
         gotoNonErrorState(CONNECTING);
-        runnable = startNewTransport();
+        startNewTransport();
```

Comment:
I'm still concerned that it is non-obvious this adds to the `channelExecutor`, like my concern in the previous PR.